### PR TITLE
JBTM-3964 test for parallel recovery scans

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-3964

!JACOCO

I'd like to be able to call periodic recovery on the LRARecoveryModule from the WildFly microprofile-lra subsystem to help with fixing issue [WFLY-20026](https://issues.redhat.com/browse/WFLY-20026).

### CODE REVIEW

**Comment by mmusgrov** [comment]
The last push adds more parallelism to the test - it now runs LRAs and scans in parallel so we verify that LRAs can start and finish while scans are in progress.

File: coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ private synchronized void recoverTransactions() {
     }
 
     private void doRecoverTransaction(Uid recoverUid) {
-        // Retrieve the transaction status from its original process.
+        // Retrieve the transaction status from its original process // TODO remove because it is not needed
```
Comment (marcosgopen) [inline coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java:105]: Do you also want to track this TODO with a JBTM?

**Review by marcosgopen — APPROVED** [review]
LGTM, Thanks Mike!

**Review by xstefank — APPROVED** [review]

**Comment by mmusgrov** [comment]
Thanks for the review

File: coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java (hunk 1 above)
Comment (mmusgrov) [reply coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java:105]: Good plan

**Pull request merged**